sqlglot/generator.py

```diff
@@ -218,6 +219,15 @@ def sep(self, sep=" "):
     def seg(self, sql, sep=" "):
         return f"{self.sep(sep)}{sql}"
 
+    def maybe_comment(self, sql, comment):
+        if not comment or not self._comments:
+            return sql
+
+        if not self.pretty or "\n" not in comment:
+            return f"{sql} /* {comment.strip()} */"
+
+        return f"/*{comment}*/\n{sql}"
```

Comment:
wait this always puts sql behind?

Reply:
Correct -- though that's only for pretty mode and for when we don't have any newlines involved. This is how we're able to preserve the following form, instead of cluttering the output with a multiline comment that's awkwardly formatted:
```sql
/* multi
    line
    comment */
SELECT * FROM bla```